### PR TITLE
Wrap fun and with-gensyms with eval-when

### PR DESCRIPTION
**Pull request by desiderius**
This simple fix allow `chanl` to be buill with Allegro CL.

### CODE REVIEW

**Comment by adlai** [comment]
Thank you for the suggestion; please post the failure that makes this necessary.

**Comment by desiderius** [comment]
```common lisp
(ql:quickload "chanl")
```
```
attempt to call `chanl::fun' which is an undefined function.
```

when compiling `utils.lisp`.

File: src/utils.lisp [hunk 1]
```diff
@@ -44,9 +44,10 @@
           (car vals-tail) next-val       ; Must... resist... the urge...
           (cdr vals-tail) (car alist)))) ; Gaaaah!
 
-(defmacro fun (&body body)
-  "This macro puts the FUN back in FUNCTION."
-  `(lambda (&optional _) (declare (ignorable _)) ,@body))
+(eval-when (:compile-toplevel :load-toplevel :execute)
```
Comment (adlai) [inline src/utils.lisp:47]: eval-always is never a good idea, so I prefer PRs with as few of these as possible.

**Review by adlai — CHANGES_REQUESTED** [review]
lift econd to minimize count of eval-always forms

**Comment by desiderius** [comment]
Sorry, I missed the change request. Thank you for your time.

**Pull request merged**